floris/heterogeneous_map.py

```diff
@@ -0,0 +1,425 @@
+from __future__ import annotations
+
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.spatial._qhull
+from scipy.interpolate import LinearNDInterpolator, NearestNDInterpolator
+from scipy.spatial import ConvexHull
+
+from floris.core.flow_field import FlowField
+from floris.logging_manager import LoggingManager
+from floris.type_dec import NDArrayFloat
+
+
+class HeterogeneousMap(LoggingManager):
+    """
+    Class for handling heterogeneous inflow configurations when defined by wind direction
+      and wind speed.
+    Args:
+        x (NDArrayFloat): A 1D NumPy array (size num_points) of x-coordinates (meters).
+        y (NDArrayFloat): A 1D NumPy array (size num_points) of y-coordinates (meters).
+        speed_multipliers (NDArrayFloat): A 2D NumPy array (size num_wd (or num_ws) x num_points)
+            of speed multipliers.  If neither wind_directions nor wind_speeds are defined, then
+            this should be a single row array
+        wind_directions (NDArrayFloat, optional): A 1D NumPy array (size num_wd) of wind directions
+            (degrees). Optional.
+        wind_speeds (NDArrayFloat, optional): A 1D NumPy array (size num_ws) of wind speeds (m/s).
+            Optional.
+
+
+    Notes:
```

Comment:
Would it be more user friendly is speed multipliers was instead n_wds x n_wss x n_points?

Reply:
Ah, I see, that's fair. Let's leave them flat, then.